storageClient/storageClient_test.go

```diff
@@ -555,3 +555,277 @@ func TestListGroupsWithClientForClientWithGroupsReturnsSuccess(t *testing.T) {
 	}
 
 }
+func TestListGroupsWithQueenClientForClientWithGroupsFilteredByGroupNameReturnsOnlyOneGroupSuccess(t *testing.T) {
+	registrationClient := accountClient.New(e3dbClients.ClientConfig{Host: cyclopsServiceHost})
+	queenClientInfo, _, err := e3dbTest.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), cyclopsServiceHost)
+	if err != nil {
+		t.Fatalf("Error %s making new account", err)
+	}
+	queenClientInfo.Host = cyclopsServiceHost
+	queenClient := New(queenClientInfo)
+	//Insert two Group To List
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	if err != nil {
+		t.Errorf("Failed generating encryption key pair %s", err)
+		return
+	}
+	newGroup := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	newGroup2 := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	response, err := queenClient.CreateGroup(testCtx, newGroup)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", newGroup, err)
+	}
+	if response.Name != newGroup.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", newGroup.Name, response.Name, newGroup)
+	}
+	groupCreatedID1 := response.GroupID
+	response, err = queenClient.CreateGroup(testCtx, newGroup2)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", newGroup2, err)
+	}
+	if response.Name != newGroup2.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", newGroup2.Name, response.Name, newGroup2)
+	}
+	groupCreatedID2 := response.GroupID
+	listRequest := ListGroupsRequest{
+		GroupNames: []string{response.Name},
+		ClientID:   uuid.MustParse(queenClient.ClientID),
+		NextToken:  0,
+		Max:        10,
+	}
+	responseList, err := queenClient.ListGroups(testCtx, listRequest)
+	if err != nil {
+		t.Fatalf("Failed to list Group: Response( %+v) \n error %+v", responseList, err)
+	}
+	for _, groupID := range []uuid.UUID{groupCreatedID2} {
+		var found bool = false
+		for _, group := range responseList.Groups {
+			if group.GroupID == groupID {
+				found = true
+				break
```

Comment:
since you have the check below in this case you don't want to break out of your loop early